src/login.js

```diff
@@ -77,13 +77,42 @@ export const auth = {
 		if (query && query.user) {
 			try {
 				let user = JSON.parse(query.user);
-				await fsp.writeFile(path.join(dir, '/creds.json'), query.user);
+				try {
+					const creds = await this.auth.loadCreds();
+					user.keys = creds.keys || {}; // preserve any exisiting keys
+				} catch (e) {
+					// skip when login is invoked for the first time and creds.json doesn't exist
```

Comment:
Unlikely to be of critical importance, but we might want to check if e has 'creds not found' vs some other error in this catch block.